lib/active_model/serializer/adapter.rb

```diff
@@ -11,6 +12,47 @@ class Adapter
 
       attr_reader :serializer
 
+      # Only the Adapter class has these methods.
+      # None of the sublasses have them.
+      class << ActiveModel::Serializer::Adapter
+        def adapter_map
+          ADAPTER_MAP
+        end
+
+        def register(name, klass)
+          adapter_map.update(name.to_s.underscore => klass)
+        end
+
+        def get(adapter)
```

Comment:
You'll notice that `get` can take a block to handle adapter not found, and is used by `get!`

```ruby
        def get!(adapter)
          get(adapter) do |failure_message| fail ArgumentError, failure_message end
        end
```